tests/add-remove.test.js

```diff
@@ -0,0 +1,129 @@
+    // Arrange
+    const descriptionInput = document.getElementById('description');
+    descriptionInput.value = '';
+
+    // Act
+    addTask();
+
+    // Retrieve and parse the savedItems from localStorage
+    const savedItems = JSON.parse(localStorage.getItem('savedItems')) || [];
+
+    // Assert
+    expect(savedItems).toEqual([]); // Check for an empty array
+  });
+
+  test('it renders the items correctly', () => {
+    // Arrange
+    const savedItems = [
+      { id: 1, description: 'Task 1', completed: false },
+      { id: 2, description: 'Task 2', completed: true },
+    ];
+    localStorage.setItem('savedItems', JSON.stringify(savedItems));
+    // Act
+    renderItems();
+    // Assert
+    const renderedItems = document.querySelectorAll('.items');
+    expect(renderedItems.length).toBe(2);
+    const item1 = renderedItems[0];
+    expect(item1.getAttribute('data-id')).toBe('1');
+    expect(item1.querySelector('.to-do-item').value).toBe('Task 1');
+    expect(item1.querySelector('.checkbox').checked).toBe(false);
+    const item2 = renderedItems[1];
+    expect(item2.getAttribute('data-id')).toBe('2');
+    expect(item2.querySelector('.to-do-item').value).toBe('Task 2');
+    expect(item2.querySelector('.checkbox').checked).toBe(true);
+  });
+
+  test('it removes the item from the local storage and re-renders items when the trash icon is clicked', () => {
+    // Arrange
+    const savedItems = [
+      { id: 1, description: 'Task 1', completed: false },
+      { id: 2, description: 'Task 2', completed: false },
+      { id: 3, description: 'Task 3', completed: false },
+    ];
+    localStorage.setItem('savedItems', JSON.stringify(savedItems));
+    renderItems();
+    const secondItemTrashIcon = document.querySelectorAll('.trash-icon')[1];
+    // Act
+    secondItemTrashIcon.dispatchEvent(new Event('click'));
+    // Assert
+    const updatedItems = JSON.parse(localStorage.getItem('savedItems'));
+    expect(updatedItems.length - 1).toBe(2);
+    expect(updatedItems[0].description).toBe('Task 1');
+    expect(updatedItems[1].description).toBe('Task 2');
+    const renderedItems = document.querySelectorAll('.items');
+    expect(renderedItems.length - 1).toBe(2);
+  });
+  test('it updates the local storage correctly when deleting an item', () => {
+    // Arrange
+    const savedItems = [
+      { id: 1, description: 'Task 1', completed: false },
+      { id: 2, description: 'Task 2', completed: false },
+    ];
+    localStorage.setItem('savedItems', JSON.stringify(savedItems));
+
+    renderItems();
+
+    const firstItemTrashIcon = document.querySelector('.trash-icon');
+    expect(document.querySelectorAll('.items').length).toBe(2);
+
+    // Act
+    firstItemTrashIcon.dispatchEvent(new Event('click'));
+
+    // Assert
+    const updatedItems = JSON.parse(localStorage.getItem('savedItems'));
+    expect(updatedItems.length - 1).toBe(1);
+    // expect(updatedItems[0].description).toBe('Task 2');
```

Comment:
**OPTIONAL** - Please remove all commented code from you codebase, it'd help make your code clean and easier to understand. Please fix this 🙏